src/pynwb/file.py

```diff
@@ -610,6 +619,40 @@ def add_stimulus_template(self, timeseries):
         self._add_stimulus_template_internal(timeseries)
         self._update_sweep_table(timeseries)
 
+    @docval({'name': 'data', 'type': (np.array, list, tuple, pd.DataFrame, NWBContainer, ScratchData),
+             'help': 'the data to add to the scratch space'},
+            {'name': 'name', 'type': str,
+             'help': 'the name of the data. Only used when passing in np.array, list, or tuple', 'default': None},
+            {'name': 'notes', 'type': str,
+             'help': 'notes to add to the data. Only used when passing in np.array, list, or tuple', 'default': None},
+            {'name': 'table_description', 'type': str,
+             'help': 'description for table. Only used when passing in pd.DataFrame', 'default': None})
+    def add_scratch(self, **kwargs):
+        '''Add data to the scratch space'''
+        data = getargs('data', kwargs)
+        if isinstance(data, (np.array, pd.DataFrame, list, tuple)):
+            name, notes, table_description = getargs('name', 'notes', 'table_description', kwargs)
+            if name is None:
+                raise ValueError('please provide a name for scratch data')
+            if isinstance(data, pd.DataFrame):
+                data = DynamicTable.from_dataframe(df=data, name=name)
+            else:
+                data = ScratchData(name=name, data=data, notes=notes)
+        self._add_scratch(data)
+
+    @docval({'name': 'name', 'type': str, 'help': 'the name of the object to get', 'default': None},
```

Comment:
I think name should be required

Reply:
Alternatively, to resolve the name issue, you could have a ``add_scratch_dataset`` and ``add_scratch_container`` function.

Reply:
I'll fix that, that was a mistake.